src/main/java/dk/cngroup/trainings/spring/springassignment/controller/AnimalController.java

```diff
@@ -14,61 +17,102 @@
+		return new ResponseEntity<>(animalService.getAnimals(), HttpStatus.OK);
+	}
 
-    @RequestMapping(path = "/{id}", method = RequestMethod.PUT)
-    public ResponseEntity<Animal> updateAnimalById(@PathVariable("id") long id,
-                                                   @RequestBody @Valid Animal animal){
-        Optional<Animal> updatedAnimal= animalService.updateAnimalById(id, animal);
-        if(updatedAnimal!=null) //TODO replace with isPresent
-            return new ResponseEntity<>(updatedAnimal.get(), HttpStatus.OK);
-        else
-            return ResponseEntity.badRequest().build(); //TODO if you want, redesign to return 404 when not found and bad request when invalid
-    }
+	@RequestMapping(path = "/{id}", method = RequestMethod.DELETE)
+	public ResponseEntity<String> deleteAnimalById(@PathVariable("id") long id) {
+		if (animalService.deleteAnimalById(id)) {
+			return new ResponseEntity<>("Success", HttpStatus.OK);
+		} else {
+			return new ResponseEntity<>("Fail: Id not found.", HttpStatus.NOT_FOUND);
+		}
+	}
 
-    @RequestMapping(path = "/{id}", method = RequestMethod.GET)
-    public ResponseEntity<Animal> getAnimalById(@PathVariable("id") long id){
-        Optional<Animal> searchedAnimal= animalService.getAnimalById(id);
-        if(searchedAnimal!=null)
-            return new ResponseEntity<>(searchedAnimal.get(),HttpStatus.OK);
-        else
-            return ResponseEntity.notFound().build();
-    }
+	@RequestMapping(path = "/{id}", method = RequestMethod.GET)
+	public ResponseEntity<Animal> getAnimalById(@PathVariable("id") long id) {
+		Optional<Animal> searchedAnimal = animalService.getAnimalById(id);
+		if (searchedAnimal.isPresent()) {
+			return new ResponseEntity<>(searchedAnimal.get(), HttpStatus.OK);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
 
-    @RequestMapping(path = "/search/{name}", method = RequestMethod.GET)
-    public ResponseEntity<List<Animal>> getAnimalByName(@PathVariable("name") String name){
-        List<Animal> sameNameAnimals= animalService.getAnimalsByName(name);
-        if(sameNameAnimals.size()!=0)
-            return new ResponseEntity<>(sameNameAnimals, HttpStatus.OK);
-        else
-            return ResponseEntity.notFound().build();
-    }
+	@RequestMapping(path = "/name", method = RequestMethod.GET)
+	public ResponseEntity<List<Animal>> getAnimalsByName(@RequestParam("name") String name) {
+		List<Animal> sameNameAnimals = animalService.getAnimalsByName(name);
+		if (!sameNameAnimals.isEmpty()) {
+			return new ResponseEntity<>(sameNameAnimals, HttpStatus.OK);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
+
+	@RequestMapping(path = "/{id}/careTakers", method = RequestMethod.POST)
+	public ResponseEntity<CareTaker> addNewCareTakerToExistingAnimal(@PathVariable("id") long id,
+																	 @RequestBody @Valid CareTaker careTaker) {
+		if (animalService.getAnimalById(id).isPresent()) {
```

Comment:
I think that this check can be done in the service and you can throw some exception if it wa not found

Reply:
Yes, I will fix it.